From 9eea82e3c458791a45829a55b03998a7572fcb9f Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Wed, 14 Mar 2018 09:13:02 +0800
Subject: [PATCH 1313/1345] fix: arm64: dts: a7k amc: enable UTMI PHY and
 disable USB3 COMPHY

commit  5660d9c54c6805f68379548b4d09ec30763329f2 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

The Armada-7020-AMC board enables PCIe x4, which occupies COMPHY 0-3.
The issue is that comphy #1 is used by USB3, and PCIe doesn't work
in x4 mode.
Enable UTMI PHY for USB2.

Change-Id: I50cb83088b06c20242574df058a28fd51093ffc3
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/52306
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7020-amc.dts |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
index 9aef7b0b..c0ff0b4 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
@@ -85,10 +85,8 @@
 			};
 
 			usb3h0: usb3@500000 {
-				separated-phys-for-usb2-usb3;
-				phys = <&utmi0>,
-				       <&comphy0 1 COMPHY_USB3H0>;
-				phy-names = "usb2", "usb3";
+				phys = <&utmi0>;
+				phy-names = "usb";
 				status = "okay";
 			};
 
@@ -157,6 +155,10 @@
 			comphy {
 				status = "okay";
 			};
+
+			utmi0: utmi@580000 {
+				status = "okay";
+			};
 		};
 
 		pcie@0x600000 {
-- 
1.7.9.5

